nixos/modules/services/monitoring/timescale-prometheus.nix

```diff
@@ -0,0 +1,272 @@
+        default = null;
+        example = "/run/keys/timescale-prometheus.db-password";
+        description = ''
+          A file containing the password corresponding to
+          <option>db-user</option>. [TS_PROM_DB_PASSWORD]
+        '';
+      };
+
+      db-port = mkOption {
+        type = types.port;
+        default = 5432;
+        description = ''
+          The TimescaleDB port [TS_PROM_DB_PORT] (default 5432)
+        '';
+      };
+
+      db-ssl-mode = mkOption {
+        type = types.str;
+        default = "disable";
+        description = ''
+          The TimescaleDB connection ssl mode [TS_PROM_DB_SSL_MODE] (default "disable")
+          disable
+          allow
+          prefer
+          require
+          verify-ca
+          verify-full
+        '';
+      };
+
+      db-user = mkOption {
+        type = types.str;
+        default = "postgres";
+        description = ''
+          The TimescaleDB user [TS_PROM_DB_USER] (default "postgres")
+        '';
+      };
+
+      labels-cache-size = mkOption {
+        type = types.int;
+        default = 10000;
+        description = ''
+          maximum number of labels to cache [TS_PROM_LABELS_CACHE_SIZE] (default 10000)
+        '';
+      };
+
+      leader-election-pg-advisory-lock-id = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Unique advisory lock id per adapter high-availability group. Set it if you want to use leader election implementation based on PostgreSQL advisory lock. [TS_PROM_LEADER_ELECTION_PG_ADVISORY_LOCK_ID]
+        '';
+      };
+
+      leader-election-pg-advisory-lock-prometheus-timeout = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Adapter will resign if there are no requests from Prometheus within a given timeout (0 means no timeout). Note: make sure that only one Prometheus instance talks to the adapter. Timeout value should be co-related with Prometheus scrape interval but add enough slack to prevent random flips. [TS_PROM_LEADER_ELECTION_PG_ADVISORY_LOCK_PROMETHEUS_TIMEOUT] (default -1ns)
+        '';
+      };
+
+      leader-election-rest = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
+          Enable REST interface for the leader election [TS_PROM_LEADER_ELECTION_REST]
+        '';
+      };
+
+      log-level = mkOption {
+        type = types.nullOr types.str;
+        default = "debug";
+        description = ''
+          The log level to use [ "error", "warn", "info", "debug" ]. [TS_PROM_LOG_LEVEL] (default "debug")
+        '';
+      };
+
+      metrics-cache-size = mkOption {
+        type = types.nullOr types.int;
+        default = 10000;
+        description = ''
+          maximum number of metric names to cache [TS_PROM_METRICS_CACHE_SIZE] (default 10000)
+        '';
+      };
+
+      migrate = mkOption {
+        type = types.str;
```

Comment:
Why are you not using a `bool` here?

Reply:
You should use `lib.booleanToString`, or whatever the function that converts bools to their textual representation then.